### PR TITLE
Allows sending of credentials (cookies) to doorkeeper

### PR DESCRIPTION
**Pull request by dmitry-silchenko**

### CODE REVIEW

File: src/clients/base-client.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class BaseClient {
     }
 
     const serviceTokenOptions = Object.assign(
-      { baseUrl: Config.serviceTokenUrl },
+      { baseUrl: Config.serviceTokenUrl, credentials: 'include' },
```
[the remark this answers is not in the document]
Comment (dmitry-silchenko) [reply src/clients/base-client.js:38]: Are you sure? These options are for token requests, for `SharkProxy.ServiceTokenClient` initialization. Please confirm, I'm not that much in details. Additionally, below in this file I can see `requestOptions.credentials = 'same-origin'` which makes me think this library is used in both ways - in browser and in nodejs apps.

[the remark this answers is not in the document]
Comment (dmitry-silchenko) [reply src/clients/base-client.js:38]: I suspect in browser, during some call to assets (for example) service if it turned out that token is expired, `SharkProxy.ServiceTokenClient` is used to refresh it. And it needs to send cookies to update token

[the remark this answers is not in the document]
Comment (jdahlke) [reply src/clients/base-client.js:38]: Yes, I am sure. You already "fixed" the cookie problem for `src/browser/service-token.js`, which is the browser implementation to retrieve service tokens.

`SharkProxy.ServiceTokenClient` is just a proxy object and the client is different between nodeJS and browser runtimes. And the browser client was fixed by you in the other file, so this additional option `credentials: 'include'` is already a default option of this client.

* NodeJS and Browser use different index files as specified in package.json
https://github.com/infopark-customers/bima-shark-js-sdk/blob/develop/package.json#L10

* https://github.com/infopark-customers/bima-shark-js-sdk/blob/develop/shark-browser.js#L19
* https://github.com/infopark-customers/bima-shark-js-sdk/blob/develop/shark-node.js#L15

[the remark this answers is not in the document]
Comment (dmitry-silchenko) [reply src/clients/base-client.js:38]: thank you

[the remark this answers is not in the document]
Comment (dmitry-silchenko) [reply src/clients/base-client.js:38]: you are correct. I just remembered that initially I wanted this `credential` options to be passed during initialization. After refactoring passing this parameter makes no sense anymore.

**Review by jdahlke — APPROVED** [review]

**Pull request merged**